pkg/k8s/client/restConfig_provider.go

```diff
@@ -12,20 +15,43 @@ import (
 	"path/filepath"
 	"strings"
 
+	"github.com/cilium/hive/cell"
+	"github.com/cilium/hive/job"
+	"github.com/fsnotify/fsnotify"
 	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/rest"
 	"k8s.io/client-go/tools/clientcmd"
 
+	"github.com/cilium/cilium/pkg/fswatcher"
 	"github.com/cilium/cilium/pkg/lock"
+	"github.com/cilium/cilium/pkg/logging"
+	"github.com/cilium/cilium/pkg/logging/logfields"
+	"github.com/cilium/cilium/pkg/time"
 	"github.com/cilium/cilium/pkg/version"
 )
 
+var (
+	log = logging.DefaultLogger.WithField(logfields.LogSubsys, "k8s-client")
+
+	// K8sAPIServerFilePath is the file path for storing kube-apiserver service and
+	// endpoints for high availability failover.
+	K8sAPIServerFilePath = "/run/cilium/state/k8sapi_server_state.json"
```

Comment:
The path of `/run/cilium/state` is not set in stone. Would be better to use `option.Config.StateDir`